src/frontend/language_phase.nit

```diff
@@ -0,0 +1,121 @@
+# Basic support of internationalization through the generation of id-to-string tables
+module language_phase
+
+private import annotation
+private import parser_util
+import astbuilder
+import gettext
+
+redef class ToolContext
+	# Main phase of `language`
+	var language_phase: Phase = new LanguagePhase(self, null)
+end
+
+class LanguagePhase
+	super Phase
+
+	redef fun process_annotated_node(nmodule, nat) do
+		if not nat.name == "language" then return
+
+		if not nmodule isa AModuledecl then
+			toolcontext.error(nmodule.location, "Cannot use annotation language on something other than a module")
+			return
+		end
+
+		var domain = nmodule.n_name.n_id.text
+
+		if nat.n_args.length != 1 then
+			toolcontext.error(nat.location, "Error: wrong number of arguments for annotation language, expected 1, got {nat.n_args.length}")
+			return
+		end
+		var arg = nat.n_args[0]
+
+		if not arg isa AStringFormExpr then
+			toolcontext.error(nat.location, "Error: expected string argument for annotation language")
+			return
+		end
+
+		var lang_raw = arg.n_string.text
+		var lang = lang_raw.substring(1, lang_raw.length - 2)
+
+		if lang == "" then
+			toolcontext.error(nat.location, "Error: invalid value for argument of annotation, expecting non-empty string")
+		end
+
+		var foldr = nmodule.location.to_s.split("/")
+		foldr.pop
+		var foldr_path = foldr.join("/").realpath
+		foldr_path += "/languages"
+
+		if not foldr_path.file_exists then foldr_path.mkdir
+
+		var lng_foldr = foldr_path + "/{lang}"
+
+		if not lng_foldr.file_exists then lng_foldr.mkdir
+
+		var mod = nmodule.parent.as(AModule)
+
+		var vi = new StringFinder(domain, foldr_path, toolcontext)
+		vi.enter_visit(mod)
+
+		var modl = nmodule.location.file.filename.basename(".nit")
+
+		var fnm = foldr_path + "/" + modl
+		var arr = new Array[PObject]
+		for v in vi.strings.values do arr.push v
```

Comment:
I was about to suggest using an array comprehension here, but why not simply..?

~~~
var arr = vi.strings.values.to_a
~~~